src/index.ts

```diff
@@ -5,14 +5,14 @@
 
 import serverless from 'serverless-http';
 import { generateServerlessRouter } from 'fhir-works-on-aws-routing';
-import { fhirConfig, genericResources } from './config';
+import { getConfig, genericResources } from './config';
 
-const serverlessHandler = serverless(generateServerlessRouter(fhirConfig, genericResources), {
-    request(request: any, event: any) {
-        request.user = event.user;
-    },
-});
-
-export default async (event: any = {}, context: any = {}): Promise<any> => {
-    return serverlessHandler(event, context);
+export default async (event1: any = {}, context: any = {}): Promise<any> => {
+    const config = await getConfig();
+    const serverlessHandler = serverless(generateServerlessRouter(config, genericResources), {
+        request(request: any, event2: any) {
+            request.user = event2.user;
+        },
+    });
+    return serverlessHandler(event1, context);

```

Comment:
This is adding ~600ms of latency to every request since all requests are building the entire `serverlessHandler` object.

The `serverlessHandler` should only be built once(on cold starts) and then be reused by all subsequent requests. 

You can do something like:
```js
const serverlessHandler: Promise<ServerlessHttp.Handler> = asyncFnThatCallsGetConfigAndBuildsTheServerlessHandler()

export default async (event: any = {}, context: any = {}): Promise<any> => {
    return (await serverlessHandler)(event, context);
};
```

Reply:
On second thought, This requires a bit more sophisticated error handling, since if the serverlessHandler fails to build for a transient error (i.e. network error when calling SSM) you don't want to "cache" that and have all subsequent requests fail.

But the above is the basic idea

Reply:
After discussing offline, we can throw errors if we can't retrieve the Integration Transform URL from SSM. Lambda will automatically terminate the container. We won't have to worry about lambda "caching" the wrong value.
https://serverlessfirst.com/fff-function-initialisation/